### PR TITLE
[NFC][New offload model][llvm-spirv] Refactor llvm-spirv options generation for enabling correct use under new offload model

### PR DESCRIPTION
**Pull request by asudarsa**
Thanks

### CODE REVIEW

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp [hunk 1]
```diff
@@ -690,6 +690,81 @@ struct Table {
   }
 };
 
+// Add any llvm-spirv option that rely on a specific Triple in addition
```
Comment (sarnex) [inline clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:693]:
```suggestion
// Add any llvm-spirv option that relies on a specific Triple in addition
```

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp [hunk 2]
```diff
@@ -690,6 +690,81 @@ struct Table {
...
+// to user supplied options.
+// NOTE: Any changes made here should be reflected in the similarly named
+// function in clang/lib/Driver/ToolChains/Clang.cpp.
+static void
+getTripleBasedSPIRVTransOpts(const ArgList &Args,
+                             SmallVector<StringRef, 8> &TranslatorArgs,
+                             const llvm::Triple Triple) {
+  bool IsCPU = Triple.isSPIR() &&
+               Triple.getSubArch() == llvm::Triple::SPIRSubArch_x86_64;
+  // Enable NonSemanticShaderDebugInfo.200 for CPU AOT and for non-Windows
```
Comment (sarnex) [inline clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:703]: Double comment, we probably only want one right?

File: clang/test/Driver/sycl-linker-wrapper-image.cpp [hunk 3]
```diff
@@ -43,12 +43,12 @@ int main() {
 // CHECK-DAG: @SYCL_PropSetName.3 = internal unnamed_addr constant [25 x i8] c"SYCL/device requirements\00"
 // CHECK-DAG: @SYCL_PropSetName.4 = internal unnamed_addr constant [22 x i8] c"SYCL/kernel param opt\00"
 // CHECK-DAG: @__sycl_offload_prop_sets_arr.5 = internal constant [3 x %_pi_device_binary_property_set_struct] [%_pi_device_binary_property_set_struct { ptr @SYCL_PropSetName, ptr @__sycl_offload_prop_sets_arr, ptr getelementptr inbounds ([1 x %_pi_device_binary_property_struct], ptr @__sycl_offload_prop_sets_arr, i64 1, i64 0) }, %_pi_device_binary_property_set_struct { ptr @SYCL_PropSetName.3, ptr @__sycl_offload_prop_sets_arr.2, ptr getelementptr inbounds ([1 x %_pi_device_binary_property_struct], ptr @__sycl_offload_prop_sets_arr.2, i64 1, i64 0) }, %_pi_device_binary_property_set_struct { ptr @SYCL_PropSetName.4, ptr null, ptr null }]
-// CHECK-DAG: @.sycl_offloading.0.data = internal unnamed_addr constant [740 x i8] 
+// CHECK-DAG: @.sycl_offloading.0.data = internal unnamed_addr constant [772 x i8]
```
Comment (sarnex) [inline clang/test/Driver/sycl-linker-wrapper-image.cpp:46]: any idea what causes the changes here?

**Review by sarnex — COMMENTED** [review]
lgtm! minor comments

File: clang/test/Driver/sycl-linker-wrapper-image.cpp (hunk 3 above)
Comment (asudarsa) [reply clang/test/Driver/sycl-linker-wrapper-image.cpp:46]: This part of testing is evolving. As we continue to add more features to SYCL offloading support in the new offload model, we update the list of options being passed to the link/post-link tools. This in turn impacts code generation. I am not sure if we are at a stage where we can accurately specify what the wrapped image should contain. But, I would let the test exist and update it for correctness as our implementation gets closer to the final product.

Thanks

Comment (sarnex) [reply clang/test/Driver/sycl-linker-wrapper-image.cpp:46]: Sounds good, probably we are passing an argument somewhere we missed before.

**Review by sarnex — APPROVED** [review]

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp (hunk 2 above)
Comment (asudarsa) [reply clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:703]: Not something I introduced. But fixed it. Thanks

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp [hunk 4]
```diff
@@ -690,6 +690,80 @@ struct Table {
   }
 };
 
+// Add any llvm-spirv option that relies on a specific Triple in addition
```
Comment (AlexeySachkov) [inline clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:693]: The same comment should be added into clang driver codebase. For now old offloading model is a default and therefore it is more likely than someone will update that part of the code without updating `clang-linker-wrapper`.

Comment (asudarsa) [reply clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:693]: Good catch. I will add the comment. We eventually would like to 'unify' the two functions. Hopefully soon. 

Thanks

**Review by mdtoguchi — APPROVED** [review]

**Comment by asudarsa** [comment]
Hi @sarnex 

I think this is ready to be merged. 

Thanks

**Comment by sarnex** [comment]
Done!

**Pull request merged**